test/parse/doctype.test.js

```diff
@@ -44,4 +44,25 @@ describe('doctype', () => {
 			expect(doc.childNodes.length).toBe(2);
 		});
 	});
+
+	test('sets the internalSubset', () => {
+		const internalSubset = `
+  <!ENTITY foo "foo">
+  <!ENTITY bar "bar">
+  <!ENTITY bar "bar2">
+  <!ENTITY % baz "baz">
+`;
+
+		const doctypeString = `
+		<?xml version="1.0">
+		<!-- >'" -->
+		<!DOCTYPE name PUBLIC "identifier" "url" [${internalSubset}]>
+		<name/>
+		`;
+
+		const { parser } = getTestParser();
+		const doc = parser.parseFromString(doctypeString, MIME_TYPE.XML_TEXT);
+
+		expect(doc.doctype).toHaveProperty('internalSubset', internalSubset);
```

Comment:
I formatted the XML snippest in a different way and removed redundant assertions that are already checked in other tests in the same file.
I also slightly changed the assertion to improve the error message in case of a failure.